File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/repository/QueryFromResourceRepository.java
package ru.tinkoff.kora.database.common.annotation.processor.repository;

import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.jdbc.JdbcRepository;

@Repository
public interface QueryFromResourceRepository extends JdbcRepository {
    @Query("classpath:/sql/test-query.sql")
    void test();
}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/repository/QueryFromResourceRepository.kt
package ru.tinkoff.kora.database.symbol.processor.repository

import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.jdbc.JdbcRepository

@Repository
interface QueryFromResourceRepository : JdbcRepository{
    @Query("classpath:/sql/test-query.sql")
    fun test()
}
